Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -114,53 +115,72 @@ std::unique_ptr<MolSupplier> getSupplier(const std::string& path,
 #endif
   }
 
+#ifdef RDK_BUILD_THREADSAFE_SSS
+  FileParsers::MultithreadedMolSupplier::Parameters params;
+  params.numWriterThreads = getNumThreadsToUse(opt.numWriterThreads);
+#endif
   //! Dispatch to the appropriate supplier
   if (fileFormat == "sdf") {
+    FileParsers::MolFileParserParams parseParams;
+    parseParams.sanitize = opt.sanitize;
+    parseParams.removeHs = opt.removeHs;
+    parseParams.strictParsing = opt.strictParsing;
 #ifdef RDK_BUILD_THREADSAFE_SSS
-    if (opt.numWriterThreads > 0) {
-      MultithreadedSDMolSupplier* sdsup = new MultithreadedSDMolSupplier(
-          strm, true, opt.sanitize, opt.removeHs, opt.strictParsing,
-          opt.numWriterThreads);
-      std::unique_ptr<MolSupplier> p(sdsup);
+    if (params.numWriterThreads > 1) {
+      auto sdsup = new FileParsers::MultithreadedSDMolSupplier(
+          strm, true, params, parseParams);
+      std::unique_ptr<FileParsers::MolSupplier> p(sdsup);
```

Comment:
This might be a case for make_unique

```
return std::make_unique<FileParsers::MultithreadedSDMolSupplier>(
    stem, true, params, parseParams);
```

Reply:
I don't think this works: we need to return a `std::unique_ptr<FileParsers::MolSupplier>`

Reply:
Correct, maybe I'll make a standards request for

```
std::make_unique<Base,Child>(a,b,c);
```

Reply:
I take it back, it does work here since you are returning a unique_ptr from the base class:

```
#include <memory>

struct Bar {
};

struct Foo:public Bar {
};

std::unique_ptr<Bar> make_foo() {
  return std::make_unique<Foo>();
}

int main() {
  auto p = make_foo();
  return 0;
}
```

Reply:
I'll give it a try

Reply:
@bp-kelley the PR for this is #7771